### PR TITLE
Mention soft delete requirement in logical replication docs

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
see commit

we already have soft-delete mention in the [alter-subscription docs](https://github.com/crate/crate/blob/49f456c9073d00f6d390704a2cb31a2d186e1ff9/docs/sql/statements/alter-subscription.rst) - I added a bit of clarification there and also mentioned the requirement in the general section

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
https://jenkins.crate.io/job/CrateDB/job/crate_on_pr/109/testReport/junit/io.crate.integrationtests/LogicalReplicationITest/test_drop_subscription/ looks like flakiness?

```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=2527, name=cratedb[subscriberm1][logical_replication][T#2], state=RUNNABLE, group=TGRP-LogicalReplicationITest]
	at __randomizedtesting.SeedInfo.seed([4DEE3971884000DB:472F02E9F40D4FE5]:0)
Caused by: org.elasticsearch.transport.NoSuchRemoteClusterException: no such remote cluster: [sub1]
```

couldn't reproduce locally by 
`./gradlew :server:test -Dtests.seed=4DEE3971884000DB -Dtests.class=io.crate.integrationtests.LogicalReplicationITest -Dtests.method="test_drop_subscription" -Dtests.locale=es-HN -Dtests.timezone=America/Toronto`

File: docs/admin/logical-replication.rst [hunk 1]
```diff
@@ -50,6 +50,13 @@ A publication is created using the
 altered or dropped using corresponding commands. The individual tables can be
 added and removed dynamically using ALTER PUBLICATION.
 
+.. CAUTION::
+
+    Publishing cluster must have
+    :ref:`setting <sql-create-table-soft-deletes-retention-lease-period>`
+    enabled so that subscribing cluster can catch up all changes made during
+    replication pause caused by network issues or explicitly done by a user.
```
Comment (seut) [inline docs/admin/logical-replication.rst:58]: The soft-delete feature is essential for the replication to work correctly for deleted documents. The lease period should not be larger than the `replication.logical.reads_poll_duration` to get all in-between changes. So its not only related to replication pause.

Comment (BaurzhanSakhariev) [reply docs/admin/logical-replication.rst]: I rephrased a bit - should I also add a sentence about `lease<=poll_duration`, clarifying it a little bit? Smth like if `poll<lease` then poll will see only part of retention

Comment (seut) [reply docs/admin/logical-replication.rst]: the lease should be `>=poll_duration`. a retention lease blocks the removal of (soft) deletes docs from lucene. once deleted docs are removed, the deletion cannot be replicated.

File: docs/admin/logical-replication.rst [hunk 2]
```diff
@@ -50,6 +50,15 @@ A publication is created using the
...
 added and removed dynamically using ALTER PUBLICATION.
 
+.. CAUTION::
+
+    Publishing cluster must have
+    :ref:`setting <sql-create-table-soft-deletes-enabled>`
+    set to ``true`` so that subscribing cluster can catch up all changes made
+    during replication pause caused by network issues or explicitly done by a user.
+    Also, :ref:`setting <sql-create-table-soft-deletes-retention-lease-period>`
+    should not be greater than ``replication.logical.reads_poll_duration``.
```
Comment (seut) [inline docs/admin/logical-replication.rst:60]:
```suggestion
    should be greater than ``replication.logical.reads_poll_duration``.
```

Comment (BaurzhanSakhariev) [reply docs/admin/logical-replication.rst:60]: or equal?

File: docs/admin/logical-replication.rst (hunk 1 above)
Comment (BaurzhanSakhariev) [reply docs/admin/logical-replication.rst]: thx for the clarification!

File: docs/admin/logical-replication.rst (hunk 2 above)
Comment (BaurzhanSakhariev) [reply docs/admin/logical-replication.rst]: added equal in https://github.com/crate/crate/pull/12138/commits/4eb76d902581bf10da6bc0fcb6b05f8d77065b3f

**Review by seut — APPROVED** [review]
👍

**Pull request merged**